presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -4325,6 +4325,9 @@ public void testLargeIn()
         assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (" + longValues + ")");
         assertQuery("SELECT orderkey FROM orders WHERE orderkey NOT IN (" + longValues + ")");
 
+        assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (mod(1000, orderkey), " + longValues + ")");
```

Comment:
@electrum Notice if you would change `longValues` to be created from `range(0, 1000)` then you will get similar exception as here: https://gist.github.com/anonymous/0ded8e4944ba14cd674d1baf6dc1c728

Reply:
I don't understand this comment. Can you explain further?

Reply:
Now everything is passing, disregarding `range(0, 5000)` or `range(0, 1000)`. But before this change with `range(0, 1000)` you will get an exception in compiler (like the one you sent me), while with `range(0, 5000)` Presto fails earlier with `StackOverflowException`.